internal/provider/networks_wireless_ssids_resource.go

```diff
@@ -0,0 +1,1727 @@
+		inner.SetPort(int32(server.Port.ValueInt64()))
+		adServers = append(adServers, *inner)
+	}
+	ad.SetServers(adServers)
+	var adCredentialsObject ADCredential
+	activeDirectoryObject.Credentials.As(ctx, &adCredentialsObject, basetypes.ObjectAsOptions{})
+	adCredentials := openApiClient.NewUpdateNetworkWirelessSsidRequestActiveDirectoryCredentials()
+	adCredentials.SetPassword(adCredentialsObject.Password.ValueString())
+	adCredentials.SetLogonName(adCredentialsObject.LogonName.ValueString())
+	ad.SetCredentials(*adCredentials)
+	request.SetActiveDirectory(*ad)
+
+	var dnsRewriteObject DNSRewrite
+	ssid.DnsRewrite.As(ctx, &dnsRewriteObject, basetypes.ObjectAsOptions{})
+	dnsRewrite := openApiClient.NewUpdateNetworkWirelessSsidRequestDnsRewrite()
+	dnsRewrite.SetEnabled(dnsRewriteObject.Enabled.ValueBool())
+	var dnsServers []string
+	for _, server := range dnsRewriteObject.DnsCustomNameservers.Elements() {
+		dnsServers = append(dnsServers, server.String())
+	}
+	dnsRewrite.SetDnsCustomNameservers(dnsServers)
+	request.SetDnsRewrite(*dnsRewrite)
+
+	var speedBurstObject SpeedBurst
+	ssid.SpeedBurst.As(ctx, &speedBurstObject, basetypes.ObjectAsOptions{})
+	speedBurst := openApiClient.NewUpdateNetworkWirelessSsidRequestSpeedBurst()
+	speedBurst.SetEnabled(speedBurstObject.Enabled.ValueBool())
+	request.SetSpeedBurst(*speedBurst)
+
+	var greObject Gre
+	ssid.Gre.As(ctx, &greObject, basetypes.ObjectAsOptions{})
+	gre := openApiClient.NewUpdateNetworkWirelessSsidRequestGre()
+	gre.SetKey(int32(greObject.Key.ValueInt64()))
+	var greConentratorObject Concentrator
+	greObject.Concentrator.As(ctx, &greConentratorObject, basetypes.ObjectAsOptions{})
+	concentrator := openApiClient.NewUpdateNetworkWirelessSsidRequestGreConcentrator(greConentratorObject.Host.ValueString())
+	gre.SetConcentrator(*concentrator)
+	request.SetGre(*gre)
+
+	var oauthObject Oauth
+	ssid.Oauth.As(ctx, &oauthObject, basetypes.ObjectAsOptions{})
+	oauth := openApiClient.NewUpdateNetworkWirelessSsidRequestOauth()
+	oauthDomains := []string{}
+	for _, domain := range oauthObject.AllowedDomains.Elements() {
+		oauthDomains = append(oauthDomains, domain.String())
+	}
+	oauth.SetAllowedDomains(oauthDomains)
+	request.SetOauth(*oauth)
+
+	var sponsorDomains []string
+	for _, domain := range ssid.SplashGuestSponsorDomains.Elements() {
+		sponsorDomains = append(sponsorDomains, domain.String())
+	}
+	request.SetSplashGuestSponsorDomains(sponsorDomains)
+	var gardenRanges []string
+	for _, gardenRange := range ssid.WalledGardenRanges.Elements() {
+		gardenRanges = append(gardenRanges, gardenRange.String())
+	}
+	request.SetWalledGardenRanges(gardenRanges)
+	var availabilityTags []string
+	for _, availabilityTag := range ssid.AvailabilityTags.Elements() {
+		availabilityTags = append(availabilityTags, availabilityTag.String())
+	}
+	request.SetAvailabilityTags(availabilityTags)
+
+	radiusServers := []openApiClient.UpdateNetworkWirelessSsidRequestRadiusServersInner{}
```

Comment:
In golang, declaring an empty slice using a literal like this is a bad pattern because you are allocating memory unnecessarily and it's not a nil value. I don't a big deal here but when we are unmarshalling the JSON payload it's the difference between a null value or an empty array.

This is preferred: 

```
var radiusServers []openApiClient.UpdateNetworkWirelessSsidRequestRadiusServersInner
```